onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1992,6 +1992,95 @@ with the exception that numpy default keepdims to False instead of True.)DOC")
       .FillUsing(QLinearMathDocGenerator("multiplication",
                                          "C = ((A - A_zero_point) * (B - B_zero_point)) * (A_scale * B_scale)/C_scale + C_zero_point"));
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(QLinearGlobalAveragePool)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(R"DOC(
+QLinearGlobalAveragePool consumes an input tensor X and applies Average pooling across
+the values in the same channel. This is equivalent to AveragePool with kernel size
+equal to the spatial dimension of input tensor. Input is of type uint8_t or int8_t.
+)DOC")
+      .Attr(
+          "nchw",
+          "Input layout, nchw or nhwc, 1 for nchw, 0 for nhwc.",
+          AttributeProto::INT)
+      .Input(
+          0,
+          "X",
+          "Input data tensor from the previous operator; "
```

Comment:
I was lazy and didn't put these comments into com.ms.QLinearConv, but if you need to keep them, the part about the image size is not accurate if you have channels last.

Reply:
Right. Add C last case.